src/grt/src/Pin.cpp

```diff
@@ -189,6 +189,18 @@ bool Pin::isDriver()
   return type == odb::dbIoType::OUTPUT || type == odb::dbIoType::INOUT;
 }
 
+odb::Point Pin::getPositionNearInstEdge(const odb::Rect& pin_box) const
+{
+  odb::Point pin_pos;
+  if (getEdge() == PinEdge::north || getEdge() == PinEdge::east) {
+    pin_pos = odb::Point(pin_box.xMax(), pin_box.yMax());
+  } else if (getEdge() == PinEdge::south || getEdge() == PinEdge::west) {
+    pin_pos = odb::Point(pin_box.xMin(), pin_box.yMin());
+  }
```

Comment:
Why does it snap in both directions?  For example on north wouldn't you just snap Y but not X?

Reply:
I made it this way to simplify the code. Since we're working only with one pin geometry at a time, I decided to get the extremes of the geometry for north/east and south/west.

I didn't think in a better way to initialize this position before, but I now noticed that we can use the pin center as the first value, and then snap only in the necessary direction. I will update it.